rho/ansible_utils.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+"""Utilities for wrapping Ansible."""
+
+from __future__ import print_function
+
+import time
+
+from getpass import getpass
+import pexpect
+import yaml
+
+from rho import utilities
+from rho.utilities import log, ANSIBLE_LOG_PATH, str_to_ascii, iteritems
+
+
+def auth_as_ansible_host_vars(auth):
+    """Get the Ansible host vars that implement an auth.
+
+    :param auth: the auth. A dictionary with fields 'id', 'name',
+        'username', 'password', and 'ssh_key_file'.
+
+    :returns: a dict that can be used as the host variables in an
+        Ansible inventory.
+    """
+
+    username = auth.get('username')
+    password = auth.get('password')
+    ssh_key_file = auth.get('ssh_key_file')
+    sudo_password = auth.get('sudo_password')
+
+    ansible_vars = {}
+
+    ansible_vars['ansible_user'] = str_to_ascii(username)
+    if password:
+        ansible_vars['ansible_ssh_pass'] = str_to_ascii(password)
+    if ssh_key_file:
+        ansible_vars['ansible_ssh_private_key_file'] = \
+            str_to_ascii(ssh_key_file)
+    if sudo_password:
+        ansible_vars['ansible_become_pass'] = sudo_password
+
+    return ansible_vars
+
+
+def redact_dict(redact_key_list, a_dict):
+    """Redact_values in a dictionary
+
+    :param redact_key_list: keys in dictionary that value should be redacted
+    :param a_dict: A dictionary to redact
+    :returns: the redacted dictionary
+    """
+    for key in redact_key_list:
+        if a_dict is not None and key in a_dict:
+            a_dict[key] = utilities.PASSWORD_MASKING
```

Comment:
This method is written to redact any general set of key:value pairs, but the constant is named `utilities.PASSWORD_MASKING` which is very specific as to what is it masking. I would suggest it be named something more general. maybe `utilities.SENSITIVE_DATA_MASK` or maybe `utilities.REDACTED_DATA_MASK`

Reply:
That variable has a bunch of other uses, all of which are for passwords, so I'd like to leave it as-is for now.

In general I think you're right, but I also don't want to add too many changes to a PR that is mostly about moving code around. In my experience, mixing refactoring and real code changes makes it much less clear what's going on.

I realize that refactors are also a natural time to spot underlying issues in the code, and I'm glad you're doing that. Please do file issues for any bugs you see, but I would only want to do very small changes as part of the refactor.

Reply:
Ok I see what you mean about this PR just being for moving code around.

One thing I've done in the past if I want to use a constant but it has a confusing name in the context I'm using it in is deal with this in a in import, i.e. 
```python
from shared_constants import CONSTANT as MY_CUSTOM_NAME
```